shared/lib_battery.h

```diff
@@ -573,48 +573,50 @@ class thermal_t
 {
 public:
 	thermal_t();
-	thermal_t(double dtHour, double mass, double length, double width, double height,
-		double Cp, double h, 
-		std::vector<double> T_room,
-		const util::matrix_t<double> &cap_vs_temp);
+	thermal_t(double dt_hour, double mass, double length, double width, double height, double R, double Cp,
+              double h, std::vector<double> T_room_K, const util::matrix_t<double> &c_vs_t);
 
-	// deep copy
+    thermal_t(const thermal_t& thermal);
```

Comment:
are lines 579, 583, and 586 duplicative? why getting rid of the const if it will be cloned/copied?

Reply:
583 and 586 are the original functions for making a copy, and I added 579. In the refactoring, I'm getting rid of 583 ad 586. 

getting rid of the const?

Reply:
gotcha, makes sense then if you'll be getting rid of 583 and 586. Isn't 579 casting what was passed in as a const thermal_t to not be const anymore, or am I misunderstanding that (highly likely)?

Reply:
class_name(const class_name& ) is the signature for the copy constructor, so it's how you would construct a new class_name from an existing one:

class_name a;
class_name b(a); 

In the copy constructor, you define how all the attributes of the class should be copied over. Some attributes can be done trivially (like if they are doubles) but if they are another class or a pointer, you'll need to specify how to make that copy.

Reply:
Good question though, thanks for all your questions so far!

Reply:
Thanks for the detailed explanations :)